.github/actions/update_submodules/action.yml

```diff
@@ -0,0 +1,10 @@
+name: "Update Submodules"
+runs:
+  using: composite
+  steps:
+    - name: "update submodules"
+      run: |
+        python --version
+        git submodule init
+        git submodule update --remote
+      shell: bash -ileo pipefail {0}
```

Comment:
nit (also applies to the other files being added):

Not sure how editors didn't add these automagically, but GitHub is flagging all of the files as not ending with a newline.